src/AutoRest.CSharp/Mgmt/AutoRest/MgmtTarget.cs

```diff
@@ -235,7 +236,17 @@ public static async Task ExecuteAsync(GeneratedCodeWorkspace project)
             //if (_overriddenProjectFilenames.TryGetValue(project, out var overriddenFilenames))
             //    throw new InvalidOperationException($"At least one file was overridden during the generation process. Filenames are: {string.Join(", ", overriddenFilenames)}");
 
-            var modelsToKeep = Configuration.MgmtConfiguration.KeepOrphanedModels.ToImmutableHashSet();
+            var emitContext = Configuration.MgmtConfiguration.IsArmCore ? Configuration.Namespace == "Azure.ResourceManager" ? true : false : true;
+            if (emitContext)
```

Comment:
well the above trenary expression is quite confusing.
Would it be possible that we change that to a helper method to make it clearer?